pkg/deploy/lattice/service_manager.go

```diff
@@ -79,13 +79,13 @@ func (m *defaultServiceManager) createServiceAndAssociate(ctx context.Context, s
 }
 
 func (m *defaultServiceManager) createAssociation(ctx context.Context, svcId *string, snName string) error {
-	sn, err := m.datastore.GetServiceNetworkStatus(snName, m.cloud.Config().AccountId)
+	snInfo, err := m.cloud.Lattice().FindServiceNetwork(ctx, snName, m.cloud.Config().AccountId)
```

Comment:
This can be a expensive call.

Reply:
Absolutely. I was planning on addressing performance at the end of the refactor, but I could potentially do it now. We were already iterating over all items in ```service_manager.go```, though this will certainly be more calls.

My thinking here was to first consolidate the logic so that we were only dealing with something more like a read-through cache, then optimize the calls. Does that sound reasonable?